src/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.Ssl.cs

```diff
@@ -283,12 +283,18 @@ protected override bool ReleaseHandle()
         private void Disconnect()
         {
             Debug.Assert(!IsInvalid, "Expected a valid context in Disconnect");
+
+            // Because we set "quiet shutdown" on the SSL_CTX, SslShutdown is supposed
+            // to always return 1 (completed success).  In "close-notify" shutdown (the
+            // opposite of quiet) there's also 0 (incomplete success) and negative
+            // (probably async IO WANT_READ/WANT_WRITE, but need to check) return codes
+            // to handle.
+            //
+            // If quiet shutdown is ever not set, see
```

Comment:
We'd have to add a shim function, but we could...

But if quiet shutdown isn't set the first call will return either 0 or -1 + WANT_READ/WANT_WRITE/WANT_ASYNC.  So we sort of have that debug-only check in the assert on 297 :)